test/acceptance/client.js

```diff
@@ -612,27 +654,15 @@ describe('The node.js OC client', function(){
         it('should contain the error details', function(){
 
           var expectedRequestWithExtraParams = {
-            url: 'http://localhost:1234',
-            method: 'post',
+            url: 'http://localhost:1234/hello-world/~1.0.0??hi=john',
```

Comment:
Yes, it is only for escaping the symbol in the regex and only in the tests. Regarding the second part - I'm not quite sure I understand what you are asking for...
If it's to add an additional test about if we set the right URL/href address then the only way this can be done in the client.js acceptance tests is to test for an error condition. There we are getting the request URL in the error message (exactly what's happening in the tests). However if we want to intercept and check the request then we should create a unit test for get-component-data because renderComponent/s methods don't us the full response, just the data in it.